android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/ConsentManager.java

```diff
@@ -0,0 +1,268 @@
+                .get(subpopulationGuid);
+    }
+
+    // if the participant's session indicates consent to this subpopulation, use that. otherwise,
+    // treat presense of consent in DAO as having consented
+    private boolean isConsentedInSessionOrLocal(UserSessionInfo session, String subpopulationGuid) {
+        ConsentStatus subpopulationStatus = getConsentStatus(subpopulationGuid);
+        if (subpopulationStatus != null && subpopulationStatus.getConsented()) {
+            return true;
+        }
+        return dao.get(subpopulationGuid) != null;
+    }
+
+    /**
+     * @return true if all required consents have been signed
+     */
+    public boolean isConsented() {
+        UserSessionInfo userSessionInfo = authenticationManager.getUserSessionInfo();
+        if (userSessionInfo != null) {
+            if (userSessionInfo.getConsented()) {
+                return true;
+            }
+
+            // Bridge session doesn't specify any consents
+            if (userSessionInfo.getConsentStatuses() == null) {
+                return true;
+            }
+
+            for (Map.Entry<String, ConsentStatus> consentStatus
+                    : userSessionInfo.getConsentStatuses().entrySet()) {
+                if (consentStatus.getValue().getRequired()
+                        && !isConsentedInSessionOrLocal(userSessionInfo, consentStatus.getKey())) {
+                    return false;
+                }
+            }
+            return true;
+        }
+        // without a user session, we can't determine whether they're consented. this shouldn't
+        // happen unless they're logged out of the application
+        return false;
+    }
+
+    /**
+     * @return true if all *required* consents have been signed and the versions signed are the most
+     * up-to-date versions of those consents
+     */
+    public boolean isConsentedMostRecent() {
+        UserSessionInfo userSessionInfo = authenticationManager.getUserSessionInfo();
+        return userSessionInfo == null ? false : userSessionInfo.getSignedMostRecentConsent();
+    }
+
+    /**
+     * @param subpopulationGuid guid for the subpopulation of the consent
+     * @param name              participant's full name
+     * @param birthdate         participant's date of birth
+     * @param base64Image       participant's signature, encoded
+     * @param imageMimeType     mime type of participant's signature
+     * @param sharingScope      participant's sharing scope for the study
+     * @return completable
+     */
+    @NonNull
+    public Completable giveConsent(@NonNull String subpopulationGuid, @NonNull String name,
+                                   @NonNull LocalDate birthdate,
+                                   @NonNull String base64Image, @NonNull String imageMimeType,
+                                   @Nullable SharingScope sharingScope) {
+        checkNotNull(subpopulationGuid);
+        checkNotNull(name);
```

Comment:
Some consent processes do not require name or birthdate, so I'm not sure whether we should checkNotNull for them.